modules/core/src/main/java/org/apache/ignite/internal/processors/performancestatistics/FilePerformanceStatisticsWriter.java

```diff
@@ -0,0 +1,529 @@
+                buf.put(strBytes);
+            }
+
+            buf.putLong(id);
+            buf.putLong(startTime);
+            buf.putLong(duration);
+            buf.put(success ? (byte)1 : 0);
+        });
+    }
+
+    /**
+     * @param type Cache query type.
+     * @param queryNodeId Originating node id.
+     * @param id Query id.
+     * @param logicalReads Number of logical reads.
+     * @param physicalReads Number of physical reads.
+     */
+    public void queryReads(GridCacheQueryType type, UUID queryNodeId, long id, long logicalReads, long physicalReads) {
+        doWrite(QUERY_READS,
+            () -> 1 + 16 + 8 + 8 + 8,
+            buf -> {
+                buf.put((byte)type.ordinal());
+                writeUuid(buf, queryNodeId);
+                buf.putLong(id);
+                buf.putLong(logicalReads);
+                buf.putLong(physicalReads);
+            });
+    }
+
+    /**
+     * @param sesId Session id.
+     * @param taskName Task name.
+     * @param startTime Start time in milliseconds.
+     * @param duration Duration.
+     * @param affPartId Affinity partition id.
+     */
+    public void task(IgniteUuid sesId, String taskName, long startTime, long duration, int affPartId) {
+        boolean needWriteStr = !stringCached(taskName);
+        byte[] strBytes = needWriteStr ? taskName.getBytes() : null;
+
+        doWrite(TASK, () -> {
+            int size = 24 + 1 + 4 + 8 + 8 + 4;
+
+            if (needWriteStr)
+                size += 4 + strBytes.length;
+
+            return size;
+        }, buf -> {
+            writeIgniteUuid(buf, sesId);
+            buf.put(needWriteStr ? (byte)1 : 0);
+            buf.putInt(taskName.hashCode());
+
+            if (needWriteStr) {
+                buf.putInt(strBytes.length);
+                buf.put(strBytes);
+            }
+
+            buf.putLong(startTime);
+            buf.putLong(duration);
+            buf.putInt(affPartId);
+        });
+    }
+
+    /**
+     * @param sesId Session id.
+     * @param queuedTime Time job spent on waiting queue.
+     * @param startTime Start time in milliseconds.
+     * @param duration Job execution time.
+     * @param timedOut {@code True} if job is timed out.
+     */
+    public void job(IgniteUuid sesId, long queuedTime, long startTime, long duration, boolean timedOut) {
+        doWrite(JOB,
+            () -> 24 + 8 + 8 + 8 + 1,
+            buf -> {
+                writeIgniteUuid(buf, sesId);
+                buf.putLong(queuedTime);
+                buf.putLong(startTime);
+                buf.putLong(duration);
+                buf.put(timedOut ? (byte)1 : 0);
+            });
+    }
+
+    /**
+     * @param op Operation type.
+     * @param sizeSupplier Record size supplier.
+     * @param writer Record writer.
+     */
+    private void doWrite(OperationType op, IntSupplier sizeSupplier, Consumer<ByteBuffer> writer) {
+        FileWriter fileWriter = this.fileWriter;
```

Comment:
I can't see why we should cache this variable.
We also, should check `enabled` flag here, but assign it to `true` as a last statement in the `start`.